src/storage/sql_table.cpp

```diff
@@ -57,6 +57,54 @@ SqlTable::SqlTable(const common::ManagedPointer<BlockStore> store, const catalog
   table_ = {new DataTable(block_store_, layout, layout_version_t(0)), layout, col_oid_to_id};
 }
 
+SqlTable::SqlTable(const common::ManagedPointer<BlockStore> store, const catalog::Schema &schema,
+                   const std::list<RawBlock *> &blocks)
```

Comment:
Two things: (1) this is 90% a copy paste job from the existing constructor, and (2) this is essentially a deep-copy constructor for a `SqlTable` object.  For (1), it would be more readable and more maintainable to have factor the common code of the two constructors into a single function (or have one constructor call the other).  For (2), I highly doubt that a deep-copy is appropriate.